galasa-managers-parent/galasa-managers-core-parent/dev.galasa.core.manager/src/main/java/dev/galasa/core/manager/internal/CoreManager.java

```diff
@@ -60,10 +74,54 @@ public void initialise(@NotNull IFramework framework, @NotNull List<IManager> al
 		}
 		this.ctf = framework.getConfidentialTextService();
 
+		if(galasaTest.isGherkin()) {
+			registerGherkinStatements(galasaTest);
+		}
+
 		// *** We always want the Core Manager initialised and included in the Test Run
 		activeManagers.add(this);
 	}
 
+	private void registerGherkinStatements(GalasaTest galasaTest) throws ManagerException {
+		for(IGherkinExecutable gherkinExecutable : galasaTest.getGherkinTest().getAllExecutables()) {
+			for(Pattern regexPattern : gherkinPatterns) {
```

Comment:
you are going to have to put a switch for the keyword here as well,  otherwise it will accept "given write to log"

Reply:
so I would put this in another static class as well